addons/hr_work_entry_contract/models/resource_calendar_attendance.py

```diff
@@ -0,0 +1,10 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class ResourceCalendarAttendance(models.Model):
+    _inherit = 'resource.calendar.attendance'
+
+    def _is_work_period(self):
+        return super()._is_work_period() and not self.work_entry_type_id.is_leave
```

Comment:
```suggestion
        return (not self.work_entry_type_id.is_leave) and super()._is_work_period()
```
If is_work_period is really long to compute and is work_entry_type_id is a leave you will do complex computation for nothing :) because in python if the first value of an AND is false the second will not be called :)